t/check_values.pl

```diff
@@ -0,0 +1,30 @@
+use Test::More;
+SKIP: { eval { require Test::Warnings; } or skip($@, 1); }
+eval { require 'ddclient'; } or BAIL_OUT($@);
+
+my @test_cases = (
+    {value => "example.com", type => 'FQDN', valid => 1},
+    {value => "example", type => 'FQDN', valid => 0},
+    {value => "https://www.example.com", type => 'URL', valid => 1},
+    {value => 'https://directnic.com/dns/gateway/ad133/', type => 'URL', valid => 1},
+    {value => 'HTTPS://MixedCase.com/', type => 'URL', valid => 1},
+    {value => 'ftp://bad.protocol/', type => 'URL', valid => 0},
+    {value => "bad-url", type => 'URL', valid => 0},
+);
+for my $tc (@test_cases) {
+    my $norm;
+    my $valid = eval {$norm = ddclient::check_value(
```

Comment:
`check_value` doesn't raise any exceptions, so there's no need to wrap it in an `eval`.

Rather than checking for valid or not, I recommend checking for the expected return value:

```perl
my @test_cases = (
    {input => 'foo', type => ddclient::T_URL, want => 'foo'},
    # ...
);
for my $tc (@test_cases) {
    my $got = ddclient::check_value($tc->{input}, ...);
    is($got, $tc->{want}, "$tc->{type}: $tc->{input}");
}
```

This would result in lots of duplication in the test case definitions (`input` and `want` will often equal each other), but it would make it easier to see what the expected behavior is, and to add normalization test cases in the future.